### DISCUSSION TITLE
Why omit getChannel ?

### DISCUSSION THREAD

**Discussion by k4lu-0p** [discordjs/discord.js#9616] [Q&A] [answered]
Hello everyone,

I'm a beginner on discord-js, especially in bot creation.
I decided to make my bot in typescript and to stay rigorous on the typing.

Below is the code for a slash command called 'join':

```ts
  data: new SlashCommandBuilder()
    .setName('join')
    .setDescription('Join a channel')
    .addChannelOption(option => option.setName('channel')
      .setDescription('The channel to join')
      .addChannelTypes(ChannelType.GuildVoice)
      .setRequired(true)
  ),
  async execute(interaction: CommandInteraction) {
    
    const voiceChannel = (interaction.options as CommandInteractionOptionResolver).getChannel('channel');
    
    // ...
}
```
Here I'm forced to cast the type, which I don't find very clean.
When I searched the definition file for an appropriate type for my interaction, all the types suffixed with Interaction omitted the 'getChannel' method (among others):

```ts
export abstract class CommandInteraction<Cached extends CacheType = CacheType> extends BaseInteraction<Cached> {
  public type: InteractionType.ApplicationCommand;
  public get command(): ApplicationCommand | ApplicationCommand<{ guild: GuildResolvable }> | null;
  public options: Omit<
    CommandInteractionOptionResolver<Cached>,
    | 'getMessage'
    | 'getFocused'
    | 'getMentionable'
    | 'getRole'
    | 'getAttachment'
    | 'getNumber'
    | 'getInteger'
    | 'getString'
    | 'getChannel'
    | 'getBoolean'
    | 'getSubcommandGroup'
    | 'getSubcommand'
  >;
  
  // ...
```
So my question is why?

thank you in advance for your reply 🙏🏻

**Comment by Idris1401** [comment, marked as answer]
didn't read just saw the code sample but try typing it as `ChatInputCommandInteraction`

**Reply by k4lu-0p** [reply]
that's it ! thanks 🙏🏻

**Comment by MrMythicalYT** [comment]
It is omitted because `CommandInteraction` is actually for *both* slash commands and context menu commands. Of course, some of these methods are only functional on one type, while the others are only functional on the other (such as `.getNumber()` which does not exist on a context menu command, or `.getMessage()` which does not exist on a slash command). You can type your interaction parameter to something more specific, `ChatInputCommandInteraction`, so it knows to allow those methods.
